src/translate/pddl_parser/parsing_functions.py

```diff
@@ -188,121 +379,271 @@ def add_effect(tmp_effect, result):
                 result.remove(contradiction)
                 result.append(new_effect)
 
-def parse_effect(alist, type_dict, predicate_dict):
+
+def parse_effect(context, alist, type_dict, predicate_dict):
+    if not alist:
+        context.error("All (sub-)effects have to be a non-empty blocks.", alist)
     tag = alist[0]
     if tag == "and":
-        return pddl.ConjunctiveEffect(
-            [parse_effect(eff, type_dict, predicate_dict) for eff in alist[1:]])
+        effects = []
+        for eff in alist[1:]:
+            if not isinstance(eff, list):
+                context.error("All sub-effects of a conjunction have to be blocks.",
```

Comment:
Why not use `expected_list_error`?

Reply:
Yeah, seems a bit inconsistent. Though, this gives clearer instructions where to look for the faulty input. would the context-layer provide enough info about this as well? Then I don't see an issue with turning this into `expected_list_error`.